internal/iptables/iptables_test.go

```diff
@@ -22,7 +23,99 @@ func TestMakeMultiportDestinations_Split(t *testing.T) {
 		[][]string{{"22:23", "25:27", "33:34", "35", "37:38", "50:54", "56", "58", "60", "63"}, {"70:72"}})
 }
 
-func assertEqual(t *testing.T, check [][]string, expected [][]string) {
+var existingRules = []byte(`# iptables-save
+# Generated by iptables-save v1.6.1 on Fri Jan  6 23:00:00 2023
+*nat
+:PREROUTING ACCEPT [0:0]
+:INPUT ACCEPT [0:0]
+:OUTPUT ACCEPT [0:0]
+:POSTROUTING ACCEPT [0:0]
+:PROXY_INIT_OUTPUT - [0:0]
+:PROXY_INIT_REDIRECT - [0:0]
+-A PREROUTING -m comment --comment "proxy-init/install-proxy-init-prerouting/testExecutionTraceID" -j PROXY_INIT_REDIRECT
+-A OUTPUT -m comment --comment "proxy-init/install-proxy-init-output/testExecutionTraceID" -j PROXY_INIT_OUTPUT
+-A PROXY_INIT_OUTPUT -o lo -m comment --comment "proxy-init/ignore-loopback/testExecutionTraceID" -j RETURN
+-A PROXY_INIT_OUTPUT -p tcp -m comment --comment "proxy-init/redirect-all-outgoing-to-proxy-port/testExecutionTraceID" -j REDIRECT --to-ports 1234
+-A PROXY_INIT_REDIRECT -p tcp -m multiport --dports 1234 -m comment --comment "proxy-init/ignore-port-1234/testExecutionTraceID" -j RETURN
+COMMIT
+# Completed on Fri Jan  6 23:00:00 2023
+`)
+
+func TestAddIncomingTrafficRules(t *testing.T) {
+	ExecutionTraceID = "testExecutionTraceID"
+
+	for _, tt := range []struct {
+		name          string
+		existingRules []byte
+		wantCommands  []*exec.Cmd
+	}{
+		{
+			name: "no existing rules, create new chain and PREROUTING rule",
+			wantCommands: []*exec.Cmd{
+				exec.Command("<iptables>", "-t", "nat", "-N", "PROXY_INIT_REDIRECT"),
+				exec.Command("<iptables>", "-t", "nat", "-A", "PROXY_INIT_REDIRECT", "-p", "tcp", "--match", "multiport", "--dports", "1234", "-j", "RETURN", "-m", "comment", "--comment", "proxy-init/ignore-port-1234/testExecutionTraceID"),
+				exec.Command("<iptables>", "-t", "nat", "-A", "PREROUTING", "-j", "PROXY_INIT_REDIRECT", "-m", "comment", "--comment", "proxy-init/install-proxy-init-prerouting/testExecutionTraceID"),
+			},
+		},
+		{
+			name:          "existing rules, flush existing chain and reuse PREROUTING rule",
```

Comment:
This formatting is odd, but I see that it's automatically happening.

Reply:
Yes, it's automatic: gofmt is aligning with the field immediately below, hence the spaces.